spock-core/src/main/java/org/spockframework/mock/runtime/ByteBuddyMockFactory.java

```diff
@@ -83,6 +83,26 @@ class ByteBuddyMockFactory {
     }
   }
 
+  /**
+   * {@return true, if the mock is considered local}
+   * <p>
+   * A mock is considered local, if all additional interfaces of the mock (including {@link ISpockMockObject}) are
+   * loadable by the target class' classloader.
+   *
+   * @param targetClass The to-be-mocked class
+   * @param additionalInterfaces Additional interfaces of the to-be-mocked type
+   */
+  @VisibleForTesting
+  static boolean isLocalMock(Class<?> targetClass, Collection<Class<?>> additionalInterfaces) {
+    // Inspired by https://github.com/mockito/mockito/blob/99426415c0ceb30e55216c3934854528c83f410e/mockito-core/src/main/java/org/mockito/internal/creation/bytebuddy/SubclassBytecodeGenerator.java#L165-L166
+    ClassLoader cl = new MultipleParentClassLoader.Builder()
```

Comment:
I would expect that you use the constructed `cl` here also as `classLoader` in line 142, if `cl != classloader`.

I your solution you "create" two classloaders, because if `cl!= classloader` then bytebuddy need to do the same sthing again.
See mockito code, which also uses the contructed classloader in that case.

Maybe change the signature to return the `cl` and then you can test against it.

Reply:
I initially opted against this, since this basically introduces another classloading strategy.

The thing I wanted to introduce was "let's gracefully fall back to the next best strategy if the Lookup strategy won't work".
Your proposal would be "Lets introduce a new strategy that is only used if the Lookup strategy is wrongfully selected and turns out to not work".

The fact that ByteBuddy would currently do the same as me is merely an implementation detail in the current version.

Imo my solution keeps the complexity lower at the small cost of an unneeded classloader construction. If you still prefer your solution, I can change the code :)

Reply:
No I think your reasoning is fine. You can leave it as it is.